### PR TITLE
fix(RELEASE-928): switch to use advisory secret

### PR DESCRIPTION
**Pull request by scoheb**
- do not get advisory repo from the ReleaseServiceConfig
- get the repo from the create-advisory secret
- see related PR https://github.com/hacbs-release/app-interface-deployments/pull/111

### CODE REVIEW

File: tasks/create-advisory/README.md [hunk 1]
```diff
@@ -11,13 +11,15 @@ Only all `redhat-pending` or all `redhat-prod` repositories may be specified in
...
-| releaseServiceConfigPath | Path to the JSON file of the ReleaseServiceConfig in the data workspace                   | No       |                             |
 | snapshotPath             | Path to the JSON file of the Snapshot spec in the data workspace                          | No       |                             |
 | dataPath                 | Path to data JSON in the data workspace                                                   | No       |                             |
 | request                  | Type of request to be created                                                             | Yes      | create-advisory             |
 | synchronously            | Whether the task should wait for InternalRequests to complete                             | Yes      | true                        |
 | pipelineRunUid           | The uid of the current pipelineRun. Used as a label value when creating internal requests | No       |                             |
 
+## Changes in 3.3.0
+- Removed releaseServiceConfig parameter as it is no longer needed.
```
Comment (johnbieren) [inline tasks/create-advisory/README.md]: `s/releaseServiceConfig/releaseServiceConfigPath/`

File: tasks/create-advisory/create-advisory.yaml
Comment (johnbieren) [inline tasks/create-advisory/create-advisory.yaml:1]: Needs version string bump

File: tasks/create-advisory/tests/test-create-advisory-fail-no-rsc.yaml
Comment (johnbieren) [inline tasks/create-advisory/tests/test-create-advisory-fail-no-rsc.yaml:1]: This whole test can be removed

File: tasks/create-advisory/README.md (hunk 1 above)
Comment (scoheb) [reply tasks/create-advisory/README.md]: fixed

File: tasks/create-advisory/create-advisory.yaml
Comment (scoheb) [reply tasks/create-advisory/create-advisory.yaml:1]: fixed

File: tasks/create-advisory/tests/test-create-advisory-fail-no-rsc.yaml
Comment (scoheb) [reply tasks/create-advisory/tests/test-create-advisory-fail-no-rsc.yaml:1]: removed

**Review by johnbieren — DISMISSED** [review]

**Comment by jinqi7** [comment]
The "create-advisories" task in rh-advisories pipeline can't pass in staging cluster. From internal-cluster console, I saw the internal pipelineRun failed with "CreateContainerConfigError".   The secret of "create-advisory-secret" can't be removed until it's not being used by staging/internal cluster.

**Comment by johnbieren** [comment]
> The "create-advisories" task in rh-advisories pipeline can't pass in staging cluster. From internal-cluster console, I saw the internal pipelineRun failed with "CreateContainerConfigError". The secret of "create-advisory-secret" can't be removed until it's not being used by staging/internal cluster.

Maybe the secret it is using needs to be updated to have `git_repo`?

**Comment by jinqi7** [comment]
> > The "create-advisories" task in rh-advisories pipeline can't pass in staging cluster. From internal-cluster console, I saw the internal pipelineRun failed with "CreateContainerConfigError". The secret of "create-advisory-secret" can't be removed until it's not being used by staging/internal cluster.
> 
> Maybe the secret it is using needs to be updated to have `git_repo`?

Has the [PR](https://github.com/hacbs-release/app-interface-deployments/pull/111) already applied in Internal-cluster?The secret is removed in the [mr](https://gitlab.cee.redhat.com/service/app-interface/-/commit/fec30e657c670b65fb413dc4ba32bb4d16015186)

**Comment by johnbieren** [comment]
> > > The "create-advisories" task in rh-advisories pipeline can't pass in staging cluster. From internal-cluster console, I saw the internal pipelineRun failed with "CreateContainerConfigError". The secret of "create-advisory-secret" can't be removed until it's not being used by staging/internal cluster.
> > 
> > 
> > Maybe the secret it is using needs to be updated to have `git_repo`?
> 
> Has the [PR](https://github.com/hacbs-release/app-interface-deployments/pull/111) already applied in Internal-cluster?The secret is removed in the [mr](https://gitlab.cee.redhat.com/service/app-interface/-/commit/fec30e657c670b65fb413dc4ba32bb4d16015186)

`create-advisory-secret` is removed in the MR. `create-advisory-stage-secret` and `create-advisory-prod-secret` still exist on the prod clusters, and `create-advisory-stage-secret` still exists on the stage cluster. The e2e should run on the stage cluster with `create-advisory-stage-secret`

**Comment by johnbieren** [comment]
> > > > The "create-advisories" task in rh-advisories pipeline can't pass in staging cluster. From internal-cluster console, I saw the internal pipelineRun failed with "CreateContainerConfigError". The secret of "create-advisory-secret" can't be removed until it's not being used by staging/internal cluster.
> > > 
> > > 
> > > Maybe the secret it is using needs to be updated to have `git_repo`?
> > 
> > 
> > Has the [PR](https://github.com/hacbs-release/app-interface-deployments/pull/111) already applied in Internal-cluster?The secret is removed in the [mr](https://gitlab.cee.redhat.com/service/app-interface/-/commit/fec30e657c670b65fb413dc4ba32bb4d16015186)
> 
> `create-advisory-secret` is removed in the MR. `create-advisory-stage-secret` and `create-advisory-prod-secret` still exist on the prod clusters, and `create-advisory-stage-secret` still exists on the stage cluster. The e2e should run on the stage cluster with `create-advisory-stage-secret`

Ah, yes that is the issue. @scoheb the secret has to be passed as a parameter. Secret name is hardcoded to `create-advisory-secret` right now https://github.com/hacbs-release/app-interface-deployments/pull/111/files#diff-05350380c5e4f51ef4f4c24e7a124fee145d8bced0b6bed0cef0e0bf919df1d5R69

**Comment by scoheb** [comment]
Created https://github.com/hacbs-release/app-interface-deployments/pull/115 to address for stage.

**Comment by scoheb** [comment]
and https://github.com/hacbs-release/app-interface-deployments/pull/116 to address prod

**Comment by scoheb** [comment]
thanks @jinqi7 and @johnbieren for finding the issue!

**Review by johnbieren — APPROVED** [review]

**Pull request merged**